### ISSUE TITLE
Framework linker error when using subspecs with dependencies.

### ISSUE THREAD

**Issue by crayment** [CocoaPods/CocoaPods#5191]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

CocoaPods version: 0.39.0

You can try the below yourself by cloning this [minimally reproducing sample project](https://github.com/crayment/SubspecTest), running `pod install`, and trying to build the example project. This project was created with `pod lib create SubspecTest` and then minimal changes were made as described below:

I have a podspec with multiple subspecs. One of these subspecs includes some helper functions for testing that is meant to only be used in test targets. Imagine in the following podspec example that  `Sub1` is meant for test targets and `Main` is meant for the application target:

```ruby
s.subspec 'Main' do |s|
  s.source_files = 'SubspecTest/Classes/Main/**/*'
  s.dependency 'Freddy'
end

s.subspec 'Sub1' do |s|
  s.source_files = 'SubspecTest/Classes/Sub1**/*'
  s.dependency 'Freddy'
end
```

As you can see these two subspecs share a dependency (in this case Freddy but doesn't matter what). In my real app `Sub1` also depends on `Main` but that seems to be irrelevant to reproducing the issue.

Now the Podfile:

```ruby
use_frameworks!

target 'SubspecTest_Example', :exclusive => true do
  pod 'SubspecTest/Main', :path => '../'
end

target 'SubspecTest_Tests', :exclusive => true do
  pod 'SubspecTest/Sub1', :path => '../'

  pod 'Quick', '~> 0.8'
  pod 'Nimble', '~> 3.0'
end
```

As I described above each target depends on a different subspec. Unfortunately this leads to the following linker error:

![](http://cl.ly/2B1m402O3Z3h/Image%202016-04-22%20at%203.08.13%20PM.png)

I just tested again and actually the `Freddy` dependency only has to exist in one of the subspecs to cause this linker error. Also removing the `SubspecTest` pod line from either target in the Podfile resolves the linker error.

Am I doing something crazy?

## What did you do?

Described above 😄 

## What did you expected to happen?

- No linker errors.
- The app target to be able to use the Main subspec code.
- The test target to be able to use the Sub1 subspec code.

## What happened instead?

Linker error. If you add an `import Freddy` in any of the pod swift files it gives you a `No such module 'Freddy'` error.

## Podfile

Included above.

**Comment by neonichu** [comment]
Could you please test this with 1.0.0.beta.8?

**Comment by crayment** [comment]
This issue does seem to be resolved in 1.0.0.beta.8.

**Comment by neonichu** [comment]
Thanks for checking, I'm going to close this, then.
